controllers/argocd/applicationset.go

```diff
@@ -758,25 +758,24 @@ func (r *ReconcileArgoCD) reconcileApplicationSetRoleBinding(cr *argoproj.ArgoCD
 }
 
 func getApplicationSetContainerImage(cr *argoproj.ArgoCD) string {
-	defaultImg, defaultTag := false, false
-
-	img := ""
-	tag := ""
-
-	// First pull from spec, if it exists
-	if cr.Spec.ApplicationSet != nil {
-		img = cr.Spec.ApplicationSet.Image
-		tag = cr.Spec.ApplicationSet.Version
-	}
 
-	// If spec is empty, use the defaults
+	defaultImg, defaultTag := false, false
+	img := cr.Spec.ApplicationSet.Image
```

Comment:
```suggestion
	img := ""
	if cr.Spec.ApplicationSet != nil {
	  img = cr.Spec.ApplicationSet.Image
	}
```
Just to be on safer side let's add a nil check. In `cr.Spec.ApplicationSet.Image`,  [ApplicationSet](https://github.com/argoproj-labs/argocd-operator/blob/f6d77ef623a2cbf4d5fe481bb891f21fad10e9ee/api/v1beta1/argocd_types.go#L743) is of type pointer, so it will  result into nil pointer exception if not set and accessed. 

Same for version aswell....

Reply:
I think it will be handled in [here](https://github.com/argoproj-labs/argocd-operator/blob/ec69f19b9fe3047c5a8109e30988051cb5a9dcf0/controllers/argocd/applicationset.go#L170-L175)
AFAIU, If appset is not enabled or no field is there under appset, it won't be executed